pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -210,9 +212,9 @@ public class PersistentTopic extends AbstractTopic implements Topic, AddEntryCal
 
     protected final MessageDeduplication messageDeduplication;
 
-    private static final long COMPACTION_NEVER_RUN = -0xfebecffeL;
+    private static final Long COMPACTION_NEVER_RUN = -0xfebecffeL;
     private CompletableFuture<Long> currentCompaction = CompletableFuture.completedFuture(COMPACTION_NEVER_RUN);
-    private final CompactedTopic compactedTopic;
+    private TopicCompactionService topicCompactionService;
```

Comment:
Should we close the compaction service when closing(deleting) the topic? We should provide a close method for the compaction service to make sure the opened resources can be closed when the topic is closed.

Reply:
There doesn't seem to be a resource that needs to be closed? , because the resource is shared in the `CompactionserviceFactory`

Reply:
But maybe the user-defined compaction service will open resources for each topic. For example, if you try to write the compacted to a new topic with a producer (each compaction task will write the compacted to a new topic and then remove the old one). When the original topic is closed, we should also close the producer for writing compacted data? And we can't close the CompactionserviceFactory because the broker is still running, just some topics move to other brokers.

Reply:
Ok, make sense.

Reply:
I open a new PR to change PIP: https://github.com/apache/pulsar/pull/20786